lib/kafo/kafo_configure.rb

```diff
@@ -293,33 +297,40 @@ def set_parameters
 
     def set_app_options
       self.class.app_option ['--[no-]colors'], :flag, 'Use color output on STDOUT',
-                            :default => !!config.app[:colors]
+                            :default => !!config.app[:colors], :advanced => true
       self.class.app_option ['--color-of-background'], 'COLOR', 'Your terminal background is :bright or :dark',
-                            :default => config.app[:color_of_background]
+                            :default => config.app[:color_of_background], :advanced => true
       self.class.app_option ['--dont-save-answers'], :flag, "Skip saving answers to '#{self.class.config.answer_file}'?",
-                            :default => !!config.app[:dont_save_answers]
+                            :default => !!config.app[:dont_save_answers], :advanced => true
       self.class.app_option '--ignore-undocumented', :flag, 'Ignore inconsistent parameter documentation',
-                            :default => false
+                            :default => false, :advanced => true
       self.class.app_option ['-i', '--interactive'], :flag, 'Run in interactive mode'
       self.class.app_option '--log-level', 'LEVEL', 'Log level for log file output',
                             :default => config.app[:log_level]
       self.class.app_option ['-n', '--noop'], :flag, 'Run puppet in noop mode?',
                             :default => false
       self.class.app_option ['-p', '--profile'], :flag, 'Run puppet in profile mode?',
-                            :default => false
+                            :default => false, :advanced => true
       self.class.app_option ['-s', '--skip-checks-i-know-better'], :flag, 'Skip all system checks', :default => false
-      self.class.app_option ['--skip-puppet-version-check'], :flag, 'Skip check for compatible Puppet versions', :default => false
+      self.class.app_option ['--skip-puppet-version-check'], :flag, 'Skip check for compatible Puppet versions', :default => false, :advanced => true
       self.class.app_option ['-v', '--verbose'], :flag, 'Display log on STDOUT instead of progressbar'
       self.class.app_option ['-l', '--verbose-log-level'], 'LEVEL', 'Log level for verbose mode output',
                             :default => 'info'
       self.class.app_option ['-S', '--scenario'], 'SCENARIO', 'Use installation scenario'
-      self.class.app_option ['--disable-scenario'], 'SCENARIO', 'Disable installation scenario'
-      self.class.app_option ['--enable-scenario'], 'SCENARIO', 'Enable installation scenario'
-      self.class.app_option ['--list-scenarios'], :flag, 'List available installation scenarios'
```

Comment:
I'm not sure this is advanced. If you consider `--scenario` as non-advanced, listing them doesn't feel advanced either.

Reply:
I struggled with these. I wanted to aim for a direction where `--help` showed the most useful, or common used set of parameters. And given we tend to pick a scenario and stick with it, I saw the scenario options as advanced. This may reduce visibility of scenarios, but for the predominant use cases this is something I'd rather revisit to focus the basic help.

Reply:
I agree that I debated them. However, I think that if you document `--scenario`, then `--list-scenarios` is also not advanced. enable/disable/force/compare are certainly advanced.

Btw, I don't know if we have the info already at this point, but perhaps we can be even smarter and see if there's more than one. If not, do not accept list options. You may want to always accept `--scenario` just so instructions like `foreman-installer --scenario foreman` does work, even if there is only one.

Reply:
I'm going to let #282 carry the conversation around the proposed change around hiding those options in some setups.